sys/kern/kcsan.c

```diff
@@ -0,0 +1,262 @@
+#include <sys/mimiker.h>
+#include <sys/vm.h>
+#include <sys/thread.h>
+#include <machine/interrupt.h>
+#include <sys/klog.h>
+
+#define WATCHPOINT_READ_BIT (1 << 31)
+#define WATCHPOINT_SIZE_SHIFT 28
+#define WATCHPOINT_SIZE_MASK (~(WATCHPOINT_ADDR_MASK | WATCHPOINT_READ_BIT))
+#define WATCHPOINT_ADDR_MASK ((1 << WATCHPOINT_SIZE_SHIFT) - 1)
+
+#define WATCHPOINT_INVALID 0
+
+#define WATCHPOINT_NUM 32
+#define SKIP_COUNT 500
+#define WATCHPOINT_DELAY 10000
```

Comment:
How was this delay chosen?

Reply:
These constants are chosen rather arbitrarily. That delay makes tests execute within a reasonable time.